core/src/main/java/com/linecorp/armeria/common/ContentDisposition.java

```diff
@@ -0,0 +1,536 @@
+     */
+    public static ContentDisposition of(String type, String name) {
+        return builder(type).name(name).build();
+    }
+
+    /**
+     * Returns a new {@link ContentDisposition} with the specified {@code type}, {@code name}
+     * and {@code filename}.
+     *
+     * @param type the disposition type like for example {@code inline}, {@code attachment},
+     *             or {@code form-data}
+     * @param name the name parameter
+     * @param filename the filename parameter that will be be formatted as quoted-string,
+     *                 as defined in RFC 2616, section 2.2, and any quote characters within
+     *                 the filename value will be escaped with a backslash,
+     *                 e.g. {@code "foo\"bar.txt"} becomes {@code "foo\\\"bar.txt"}
+     */
+    public static ContentDisposition of(String type, String name, String filename) {
+        return builder(type).name(name).filename(filename).build();
+    }
+
+    /**
+     * Returns an empty {@link ContentDisposition}.
+     */
+    public static ContentDisposition of() {
+        return EMPTY;
+    }
+
+    /**
+     * Parses a {@code "content-disposition"} header value as defined in RFC 2183.
+     *
+     * <p>Note that only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * @param contentDisposition the {@code "content-disposition"} header value
+     * @return the parsed content disposition
+     * @see #asHeaderValue()
+     */
+    public static ContentDisposition parse(String contentDisposition) throws UnsupportedEncodingException {
+        requireNonNull(contentDisposition, "contentDisposition");
+        final List<String> parts = tokenize(contentDisposition);
+        final String type = parts.get(0);
+        String name = null;
+        String filename = null;
+        Charset charset = null;
+        Long size = null;
+        ZonedDateTime creationDate = null;
+        ZonedDateTime modificationDate = null;
+        ZonedDateTime readDate = null;
+        for (int i = 1; i < parts.size(); i++) {
+            final String part = parts.get(i);
+            final int eqIndex = part.indexOf('=');
+            if (eqIndex != -1) {
+                final String attribute = part.substring(0, eqIndex);
+                final String value;
+                if (part.startsWith("\"", eqIndex + 1) && part.endsWith("\"")) {
+                    value = part.substring(eqIndex + 2, part.length() - 1);
+                } else {
+                    value = part.substring(eqIndex + 1);
+                }
+
+                if ("name".equals(attribute)) {
+                    name = value;
+                } else if ("filename*".equals(attribute)) {
+                    final int idx1 = value.indexOf('\'');
+                    final int idx2 = value.indexOf('\'', idx1 + 1);
+                    if (idx1 != -1 && idx2 != -1) {
+                        charset = Charset.forName(value.substring(0, idx1).trim());
```

Comment:
Could we keep a map of all supported encoding names instead of using `Charset.forName()`? e.g.

```java
private static final Map<String, Charset> supportedCharsets =
    ImmutableMap.of("utf-8", StandardCharset.UTF_8, ....);

charset = supportedCharsets.get(Ascii.toLowercase(value.substring(...).trim()));
checkArgument(charset != null, "Charset: %s ..."));
```